tests/trainer/test_trainer.py

```diff
@@ -518,6 +521,107 @@ def on_load_checkpoint(self, _):
         assert next_model.num_on_load_checkpoint_called == 1
 
 
+def test_trainer_properties_restore_resume_from_checkpoint(tmpdir):
+    """Test that required trainer properties are set correctly when resuming from checkpoint in different
+    phases."""
+
+    class CustomClassifModel(ClassificationModel):
+        def configure_optimizers(self):
+            optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)
+            lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
+            return [optimizer], [lr_scheduler]
+
+    model = CustomClassifModel()
+    dm = ClassifDataModule()
+    checkpoint_callback = ModelCheckpoint(dirpath=tmpdir, save_last=True)
+    trainer_args = dict(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        logger=False,
+        callbacks=[checkpoint_callback],
+        num_sanity_val_steps=0,
+    )
+    trainer = Trainer(**trainer_args)
+    trainer.fit(model, datamodule=dm)
+
+    resume_ckpt = str(tmpdir / "last.ckpt")
+    state_dict = torch.load(resume_ckpt)
+
+    trainer_args.update(
+        {"max_epochs": 3, "resume_from_checkpoint": resume_ckpt, "checkpoint_callback": False, "callbacks": []}
```

Comment:
why does this needs to pass an empty callback list?

Reply:
because trainer_args contains a checkpoint_callback inside callbacks few lines above. If I don't remove it, it will keep update last ckpt in trainer.fit will be override the original ckpt.